File: 1.0/src/cz/mg/vulkan/PFNvkMergeValidationCachesEXT.java
package cz.mg.vulkan;

public class PFNvkMergeValidationCachesEXT extends VkFunctionPointer {
    public PFNvkMergeValidationCachesEXT() {
    }

    protected PFNvkMergeValidationCachesEXT(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkMergeValidationCachesEXT(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkMergeValidationCachesEXT(long value) {
        setValue(value);
    }

    public PFNvkMergeValidationCachesEXT(VkInstance instance) {
        super(instance, new VkString("vkMergeValidationCachesEXT"));
    }

    public int call(VkDevice device, VkValidationCacheEXT dstCache, int srcCacheCount, VkValidationCacheEXT pSrcCaches){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), dstCache != null ? dstCache.getVkAddress() : VkPointer.getNullAddressNative(), srcCacheCount, pSrcCaches != null ? pSrcCaches.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long device, long dstCache, int srcCacheCount, long pSrcCaches);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkMergeValidationCachesEXT.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkMergeValidationCachesEXT_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong dstCache, jint srcCacheCount, jlong pSrcCaches){
    (void)env;
    (void)jc;
    PFN_vkMergeValidationCachesEXT f = (PFN_vkMergeValidationCachesEXT)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkValidationCacheEXT*)jniLongToPointer(dstCache)),
        (uint32_t)srcCacheCount,
        ((VkValidationCacheEXT*)jniLongToPointer(pSrcCaches))
    );
}

